core/controllers/admin.py

```diff
@@ -335,3 +336,50 @@ def get(self):
             'attachment; filename=topic_similarities.csv')
         self.response.write(
             recommendations_services.get_topic_similarities_as_csv())
+
+
+class DataExtractionQueryHandler(base.BaseHandler):
+
+    @require_super_admin
+    def post(self):
+        exp_id = self.payload.get('exp_id')
+        exp_version = self.payload.get('exp_version')
+        state_name = self.payload.get('state_name')
+        num_answers = self.payload.get('num_answers')
+
+        exploration = exp_services.get_exploration_by_id(
+            exp_id, strict=False, version=exp_version)
+
+        if exploration is None:
+            response = {
+                'success': False,
+                'message': 'no exploration with ID \'%s\' exist.' % exp_id,
+                'data': None
+            }
+            self.render_json(response)
+            return
+
+        if state_name not in exploration.states.keys():
+            response = {
+                'success': False,
+                'message': (
+                    'exploration \'%s\' does not have \'%s\' state.'
+                    % (exp_id, state_name)),
+                'data': None
+            }
+            self.render_json(response)
+            return
+
+        state_answers = stats_services.get_state_answers(
+            exp_id, exp_version, state_name)
+        extracted_answers = state_answers.get_submitted_answer_dict_list()
+
+        if num_answers > 0:
+            extracted_answers = extracted_answers[:num_answers]
+
+        response = {
```

Comment:
If we raise 400 for the error cases then just returning the extracted data should suffice.